src/main/java/org/zalando/nakadi/service/ConsumerLimitingCleaningService.java

```diff
@@ -0,0 +1,141 @@
+    private static final Logger LOG = LoggerFactory.getLogger(ConsumerLimitingCleaningService.class);
+    private static final int HANGING_NODES_CLEAN_PERIOD_H = 6;
+
+    private final ZooKeeperHolder zkHolder;
+    private final ObjectMapper objectMapper;
+    private final ConsumerLimitingService limitingService;
+
+    @Autowired
+    public ConsumerLimitingCleaningService(final ZooKeeperHolder zkHolder, final ObjectMapper objectMapper,
+                                           final ConsumerLimitingService limitingService) {
+        this.zkHolder = zkHolder;
+        this.objectMapper = objectMapper;
+        this.limitingService = limitingService;
+    }
+
+    @Scheduled(fixedRate = HANGING_NODES_CLEAN_PERIOD_H * 3600000)
+    public void cleanHangingNodes() {
+        LOG.info("Trying to run cleaning of 'hanging' connection ZK nodes");
+
+        if (!timeToRunCleaning()) {
+            return;
+        }
+        if (!createCleaningLockNode()) {
+            return;
+        }
+        try {
+            LOG.info("Cleaning of 'hanging' connection ZK nodes will be performed on this instance");
+            cleanHangingConsumersNodes();
+            updateCleaningTimestamp();
+        } catch (JsonProcessingException e) {
+            LOG.error("Json processing error when updating cleaning timestamp in ZK", e);
+        } catch (Exception e) {
+            LOG.error("Zookeeper error when cleaning consumer nodes in ZK", e);
+        } finally {
+            deleteCleaningLockNode();
+        }
+    }
+
+    private void cleanHangingConsumersNodes() throws Exception {
+        // try to remove every consumer node;
+        // the nodes that have children will fail to be removed
+        zkHolder.get()
+                .getChildren()
+                .forPath(CONNECTIONS_ZK_PATH)
+                .forEach(limitingService::deletePartitionNodeIfPossible);
+    }
+
+    private void deleteCleaningLockNode() {
+        // delete the node that is used as a lock to be
+        // captured by Nakadi instance that performs cleaning
+        try {
+            zkHolder.get()
+                    .delete()
+                    .guaranteed()
+                    .forPath(CLEANING_ZK_PATH + "/lock");
+        } catch (Exception e) {
+            LOG.error("Zookeeper error when deleting consumer cleaning lock-node", e);
+        }
+    }
+
+    private void updateCleaningTimestamp() throws Exception {
+        try {
+            // try to create node for the case if it doesn't exist
+            zkHolder.get()
+                    .create()
+                    .creatingParentsIfNeeded()
+                    .forPath(CLEANING_ZK_PATH + "/latest");
+        } catch (final KeeperException.NodeExistsException e) {
+            // "latest" node may already exist - this is ok
+        }
+        // update the latest timestamp of cleaning
+        final DateTime now = new DateTime(DateTimeZone.UTC);
```

Comment:
I would like to have this date in readable format in Zookeeper (to be able to read the date by human directly accessing ZK).